chain/epoch-manager/src/proposals.rs

```diff
@@ -270,4 +267,27 @@ mod old_validator_selection {
             rng_seed,
         ))
     }
+
+    fn shuffle_duplicate_proposals(dup_proposals: &mut Vec<u64>, rng_seed: RngSeed) {
+        let mut rng: Hc128Rng = SeedableRng::from_seed(rng_seed);
+        dup_proposals.shuffle(&mut rng);
```

Comment:
One thing I am worried here is that the specific shuffling algorithm that `rand` is using is unspecified. So in theory upgrading `rand` might change the result here. 

If we are replacing protocol_defining_rand with a fixed rng (which is :heart: ), I'd sleep better if we also vendored our own `knuth_shuffle`, so that we avoid problems a-la https://github.com/rust-lang/rust/issues/85773. 

We have tests here, so they'll probably catch *drastic* changes in the algorithm, but if it is something more subtle (like, use some super-advanced simd on specific CPUs or whatever), we might miss it.